core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsService.java

```diff
@@ -0,0 +1,113 @@
+                    nlpjsApi.getAgentInfo(agentId).request().url(), agentResponse.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
+        }
+        return agentResponse.body();
+    }
+
+
+    public boolean createAgent(AgentInit agentInit) throws IOException, IntentRecognitionProviderException {
+        Response<Void> response = nlpjsApi.createAgent(agentInit).execute();
+        if (!response.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(response.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error message \"{3}\" ",
+                    nlpjsApi.createAgent(agentInit).request().method(),
+                    nlpjsApi.createAgent(agentInit).request().url(), response.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
+        }
+        return true;
+    }
+
+    public boolean trainAgent(String agentId, TrainingData trainingData) throws IOException, IntentRecognitionProviderException {
+        Response<Void> response = nlpjsApi.trainAgent(agentId, trainingData).execute();
+        if (!response.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(response.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error message: {3} ",
+                    nlpjsApi.trainAgent(agentId, trainingData).request().method(),
+                    nlpjsApi.trainAgent(agentId, trainingData).request().url(), response.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
+        }
+        return true;
+    }
+
+    public RecognitionResult getIntent(@NonNull String agentId, @NonNull UserMessage userMessage) throws IOException, IntentRecognitionProviderException {
+        Response<RecognitionResult> response = nlpjsApi.getIntent(agentId, userMessage).execute();
+        if (!response.isSuccessful()) {
+            ErrorBody errorBody = gson.fromJson(response.errorBody().string(), ErrorBody.class);
+            String errorMessage = MessageFormat.format("Unsuccessful REST operation {0} " +
+                            "on {1}. The API responded with the status {2} and the error code {3} ",
+                    nlpjsApi.getIntent(agentId, userMessage).request().method(),
+                    nlpjsApi.getIntent(agentId, userMessage).request().url(), response.code(), errorBody.getMessage());
+            Log.error(errorMessage);
+            throw new IntentRecognitionProviderException(errorMessage);
```

Comment:
Just throw the exception, no need to log, this should be propagated to a catch that takes care of logging the error (let me know if it is not the case)

Reply:
👍